### PR TITLE
feat(metrics): send missing event metrics in Glean events ping from the frontend

### PR DESCRIPTION
**Pull request by akkomar**
This is a follow-up to https://github.com/mozilla/fxa/pull/16499 that:
1. Adds recording of remaining event metrics
2. Adds `reason` extras in `reg_engage` and `login_submit_frontend_error`

That were missed in the previous PR.

See also https://mozilla-hub.atlassian.net/browse/DENG-2407

## Because

- We want to use event metric type in Glean telemetry to better document events.

## This pull request

Adds support for sending remaining event metrics in events ping via Glean.js.

## Issue that this pull request solves

https://mozilla-hub.atlassian.net/browse/DENG-2407

## Checklist

_Put an `x` in the boxes that apply_

- [ ] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml [hunk 1]
```diff
@@ -366,6 +367,23 @@ cached_login:
     expires: never
     data_sensitivity:
       - interaction
+  forgot_pwd_submit:
+    type: event
+    description: |
+      TODO: added in https://github.com/mozilla/fxa/pull/16668
```
Comment (akkomar) [inline packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:373]: @chenba can you help me write this description?

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml [hunk 2]
```diff
@@ -875,3 +922,37 @@ reg:
...
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1830504
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1844121
+    expires: never
+    data_sensitivity:
+      - interaction
+  cwts_engage:
+    type: event
+    description: |
+      TODO: added in https://github.com/mozilla/fxa/pull/16668
```
Comment (akkomar) [inline packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:946]: @chenba can you help me write this description?

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml (hunk 1 above)
Comment (chenba) [reply packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:373]: 'User clicks on the "Forgot Password" Link.'

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml (hunk 2 above)
Comment (chenba) [reply packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:946]: 'User interacted with the Sync "Choose What to Sync" options during account registration.'

File: packages/fxa-content-server/app/scripts/lib/glean/index.ts [hunk 3]
```diff
@@ -168,6 +188,36 @@ const recordEventMetric = (eventName: string) => {
...
+      break;
+    case 'password_reset_recovery_key_view':
+      passwordReset.recoveryKeyView.record();
+      break;
+    case 'password_reset_submit':
+      passwordReset.submit.record();
+      break;
+    case 'password_reset_view':
+      passwordReset.view.record();
+      break;
```
Comment (chenba) [inline packages/fxa-content-server/app/scripts/lib/glean/index.ts:220]: Thank you for adding these.  The Backbone UI is not longer used to handle password resets so these won't be used.  But it doesn't hurt anything to leave them here.

**Review by chenba — APPROVED** [review]
@akkomar Thank you!

**Pull request merged**